terra_2_microreact/microreact_export.py

```diff
@@ -0,0 +1,342 @@
+    "columns": columns
+  }    
+
+  if tree_files:
+    tree_files_dict, tree_dict = create_project_entry("tree",id_column, remove_file_columns, None, None, None, tree_files)
+  else:
+    logger.info("No tree files provided; skipping tree entry creation.")
+    tree_files_dict, tree_dict = {}, {}
+  
+  if "latitude" in headers and "longitude" in headers:
+    map_entry = create_project_entry("map", "", None, None, None, None)
+    logger.info("Map entry created based on presence of latitude and longitude in metadata.")
+  else:
+    logger.info("No latitude/longitude fields found; skipping map entry creation.")
+    map_entry = {}
+
+  project_input = {
+    "schema": "https://microreact.org/schema/v1.json",
+    "meta": {
+      "name": project_name
+    },
+    "datasets": {
+      dataset_id: dataset_entry
+    },
+    "files": {
+      **metadata_entry,
+      **tree_files_dict
+    },
+    "tables": {
+      table_id: table_entry
+    },
+    "trees": {
+      **tree_dict
+    },
+    "charts": {},
+    "filters": {},
+    "maps": {
+      **map_entry
+    },
+    "networks": {},
+    "notes": {},
+    "panes": {},
+    "slicers": {},
+    "styles": {},
+    "timelines": {},
+    "views": {},
+    "matrices": {}
+  }
+
+  # Write Input to JSON
+  project_input_json = json.dumps(project_input)
+  logger.info(f"Created Microreact project JSON")
+
+  if access_token:
+    url = "https://microreact.org/api/projects/create"
+    if restricted_access:
+      url += "?access=private"
+    headers = {
+        "Content-Type": "application/json; charset=UTF-8",
+        "Access-Token": access_token
+    }
+    response = requests.post(url, headers=headers, json=project_input)
+    logger.info(f"Microreact API response status: {response.status_code}")
+    if response.status_code == 200:
+      microreact_response = response.json()
+      logger.info(f"Microreact project created successfully with ID: {microreact_response.get('id')}")
+    else:
+      logger.error(f"Failed to create Microreact project: {response.text}")
+
+  return microreact_response, project_input_json
+
+def update_microreact_project(
+    project_url: str,
+    access_token: str,
+    metadata: Optional[str],
+    id_column: Optional[str],
+    date_column: Optional[str],
+    remove_file_columns: bool,
+    tree_files: Optional[List[Path]]
+):
+  logger.info(f"Updating existing Microreact project at {project_url}")
+  url = f"https://microreact.org/api/projects/update?project={project_url}"
+  get_url = f"https://microreact.org/api/projects/json?project={project_url}"
+  get_request_headers = {
+    "Access-Token": access_token
+  }
+  get_response = requests.get(get_url, headers=get_request_headers)
+  updated_project = get_response.json()
+  logger.info(f"Fetched current project data for update")
+  if metadata:
+    metadata_entry, metadata_id, headers = create_project_entry("metadata", id_column, date_column, remove_file_columns, metadata, None, None, None)
```

Comment:
This function requires seven params https://github.com/theiagen/theiagen_docker_builds/pull/17/files#diff-0d9da002e743afd9d0c2f18fe13ebc530166d0740feb2dca71276a9fdce1ac54R84 but here its input eight